cls/SourceControl/Git/Utils.cls

```diff
@@ -1604,16 +1605,64 @@ ClassMethod RunGitCommandWithInput(command As %String, inFile As %String = "", O
 
     set newArgs($increment(newArgs)) = command
 
+    // defining variables for if statement use later
+    set syncIris = 0
+    set diffBase = ""
+    set diffCompare = ""
+    if (command = "checkout") || (command = "merge") || (command = "rebase") {
+        set syncIris = 1
+        set diffCompare = args(args)
+    }
+    
+    
     for i=1:1:$get(args) {
         if ($data(args(i))) {
             set newArgs($increment(newArgs)) = args(i)
+            if (args(i) = "checkout") || (args(i) = "merge") || (args(i) = "rebase") {
```

Comment:
How much effort would it be to include "pull"? It's not mentioned in the issue, but it would be nice to include pull and refactor the duplicated code out of the Pull() method on this class.